static/js/pv_tree.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { Component } from "react";
+import hierarchy from "../../tools/pv_tree_generator/hierarchy_golden.json";
+//import hierarchy from "../data/hierarchy.json";
+
+interface NodePropType {
+  title: string;
+  selected: string;
+  expanded: string;
+  count: number;
+  children: NodePropType[];
+  type: string;
+  argString: string;
+  updateUrl: (statvar: string, add: boolean) => void;
+  nodePath: string;
+}
+
+interface NodeStateType {
+  checked: boolean;
+  expanded: boolean;
+  nodePath: string;
+}
+
+class Node extends Component<NodePropType, NodeStateType> {
+  constructor(props: NodePropType) {
+    super(props);
+    this._handleCheckboxClick = this._handleCheckboxClick.bind(this);
+    this._handleExpandClick = this._handleExpandClick.bind(this);
+    this.state = {
+      checked: props.selected === "yes" ? true : false,
+      expanded: props.selected === "yes" ? true : false,
+      nodePath: props.nodePath + "," + props.title,
+    };
+  }
+
+  render() {
+    let checkboxImg;
```

Comment:
good to have type like: let checkboxImg: HTMLInputElement for all the let varibles

Reply:
done.